### PR TITLE
add execution error log proto definition

### PR DESCRIPTION
**Pull request by koseoyoung**

### CODE REVIEW

File: proto/api/api.proto [hunk 1]
```diff
@@ -16,7 +16,7 @@ import "proto/types/action.proto";
 import "proto/types/blockchain.proto";
 import "proto/types/node.proto";
 import "proto/types/election.proto";
-import "proto/types/transaction_log.proto";
+import "proto/types/system_log.proto";
```
Comment (dustinxie) [inline proto/api/api.proto]: let's not change the file name? to be consistent with iotex-core, where it is also named TransactionLog

File: proto/api/api.proto [hunk 2]
```diff
@@ -181,6 +181,7 @@ message ActionInfo {
 message ReceiptInfo {
   iotextypes.Receipt receipt = 1;
   string blkHash = 2;
+  string executionErrorMessage = 3;
```
Comment (dustinxie) [inline proto/api/api.proto]: it seems most straightforward to me that we could just add `string executionError` to iotextypes.Receipt?

1. we only need to store 1 revert error message, and `iotextypes.Receipt` already contains action hash
2. by doing so we don't need to change api.proto and transaction_log.proto below
3. and we don't even need to change api in iotex-core -- it's just that the Receipt message now contains an extra field for front-end to parse and display

Comment (dustinxie) [reply proto/api/api.proto]: i see why you add to ReceiptInfo, that is, if we add it to iotextypes.Receipt, is the hash compatible with current protobuf, with executionError = empty?
```go
hash( newReceipt(a, b, c ... "") == hash( oldReceipt(a, b, c ...) )?
```
could you write a test to verify this? like with current protobuf
```go
receipt := &Receipt{1, 1, hash.ZeroHash256, 1, "test", []*Log{testLog}, nil}
h := receipt.Hash()
```
and with new protobuf, `string executionError` added to iotextypes.Receipt
```go
receipt := &Receipt{1, 1, hash.ZeroHash256, 1, "test", []*Log{testLog}, nil, ""} // empty executionError
h1 := receipt.Hash()
```
check if h1 == h

Comment (koseoyoung) [reply proto/api/api.proto]: I verified this working well

Comment (koseoyoung) [reply proto/api/api.proto]: image

File: proto/types/action.proto [hunk 3]
```diff
@@ -262,6 +262,7 @@ message Receipt {
   uint64 gasConsumed = 4;
   string contractAddress = 5;
   repeated Log logs = 6;
+  string executionErrorLog = 7;
```
Comment (dustinxie) [inline proto/types/action.proto]: nit: executionErrorMsg, or revertErrorMsg?

**Review by dustinxie — APPROVED** [review]

File: proto/types/action.proto (hunk 3 above)
Comment (koseoyoung) [reply proto/types/action.proto]: How about executionRevertMsg?

**Pull request merged**